tests/unit_tests/test_dqc_us_0004.py

```diff
@@ -108,19 +107,53 @@ def test_bv_None_context(self):
         modelXbrl.factsByQname = mock_facts_by_qname
 
         error_count = 0
-        for asset, liability, date in _assets_eq_liability_equity(modelXbrl):
+        for asset, liability, date in dqc_us_0004._assets_eq_liability_equity(modelXbrl):
             error_count += 1
         self.assertEqual(error_count, 0)
 
     def test_values_unequal_equal_values(self):
-        self.assertFalse(_values_unequal(100, 100, -1))
-        self.assertFalse(_values_unequal(130, 100, -2))
+        self.assertFalse(dqc_us_0004._values_unequal(100, 100, -1))
+        self.assertFalse(dqc_us_0004._values_unequal(130, 100, -2))
 
     def test_values_unequal_unequal_but_still_equal_values(self):
-        self.assertFalse(_values_unequal(120, 100, -1))
-        self.assertFalse(_values_unequal(200, 100, -2))
+        self.assertFalse(dqc_us_0004._values_unequal(120, 100, -1))
+        self.assertFalse(dqc_us_0004._values_unequal(200, 100, -2))
 
     def test_values_unequal_very_unequal_values(self):
-        self.assertTrue(_values_unequal(220, 100, -1))
-        self.assertTrue(_values_unequal(600, 100, -2))
-
+        self.assertTrue(dqc_us_0004._values_unequal(220, 100, -1))
+        self.assertTrue(dqc_us_0004._values_unequal(600, 100, -2))
+
+    def test_values_scale_nan(self):
+        """
+        Make sure that the scale value is not valid if it is Not a Number
+        """
+        self.assertFalse(dqc_us_0004._min_dec_valid(float('nan')))
+        self.assertFalse(dqc_us_0004._min_dec_valid(float('nan')))
```

Comment:
This and the next few tests are running the same check twice.  I'd recommend adding a case that you expect to return True and a case that you'd expect to return False.  This is just testing 'nan' twice.  Same applies to 'inf' and '-inf' below.

Looking good otherwise!